### ISSUE TITLE
OpenMPI 4.1.0 fails with [external] pmix 4.0.0

### ISSUE THREAD

**Issue by amckinstry** [open-mpi/ompi#8335]
Thank you for taking the time to submit an issue!

## Background information

OpenMPI 4.1.0
PMIX 4.0.0

Debian Unstable (and ubuntu)

When built against external 4.0.0 , we see segfaults on simple hello world tests:
```
autopkgtest [06:32:43]: test hello2: [-----------------------
[ci-359-a03b0931:01932] *** Process received signal ***
[ci-359-a03b0931:01932] Signal: Segmentation fault (11)
[ci-359-a03b0931:01932] Signal code: Invalid permissions (2)
[ci-359-a03b0931:01932] Failing at address: 0x5632cd610c70
[ci-359-a03b0931:01932] [ 0]
/lib/x86_64-linux-gnu/libc.so.6(+0x3bd00)[0x7fcc11a84d00]
[ci-359-a03b0931:01932] [ 1] [0x5632cd610c70]
[ci-359-a03b0931:01932] *** End of error message ***
Segmentation fault
autopkgtest [06:32:46]: test hello2: -----------------------]
autopkgtest [06:32:46]: test hello2:  - - - - - - - - - - results - -
- - - - - - - -
```

For the moment we have reverted to internal pmix (3.2.2?)

**Comment by jsquyres** [comment]
@rhc54 @jjhursey Have you guys heard about this before, perchance?

@amckinstry Can you send us the full build details?  E.g., `./configure` line for both PMIx and Open MPI.

**Comment by rhc54** [comment]
@amckinstry I looked into this and found the issue - a configure code that is looking for `major version == 3` instead of `major version >= 3`. Should be a trivial change - @jjhursey said he would fix it later today. Thanks for the report and sorry for the problem.

**Comment by jjhursey** [comment]
Thanks for letting us know. I filed fixes for the two latest release streams:
 * `v4.0.x` branch in PR #8338
 * `v4.1.x` branch in PR #8337
 * This will need to go back to PRRTE as well. I'll update this comment with the PR link once I file it.

**Comment by amckinstry** [comment]
Thanks.  I can now build openmpi 4.1.0 with the external pmix 4.0 with this patch, but still have issues due to a bug in pix 4.0.0 ( https://github.com/open-mpi/ompi/issues/8323 ).

Because of this, openmpi in Debian Unstable is currently configured to use the internal pmix 3.* but appears to be fragile in the presence of pmix 4.0 on the system ( mpich and slurm are also configured to use the external pmix).

(pmix is built as a shared library)

**Comment by rhc54** [comment]
> appears to be fragile in the presence of pmix 4.0

@amckinstry Can you please expand on this statement? Are you just referring to https://github.com/openpmix/openpmix/issues/2003 or is there some other problem?

**Comment by amckinstry** [comment]
Ok, with an apparent fix i've now reverted to openmpi 4.1.0 usng external pmix 4.0.0 on Debian Unstable.
By "fragile" I mean if openmpi is built to use internal pmix, while another copy of pmix (both libpmix.so.2) exists on the system, albeit with different paths, bugs are likely to ensue - picking up the wrong library at build or runtime.

I'm not sure that openmpi/openpmix should be doing anything (more) about this - I think its a Debian issue to ensure that only one libpmix.so.2 exists on the machine at a time.

**Comment by jsquyres** [comment]
Merged into the v4.1.x branch -- fixed!
